internal/component/common/loki/positions/positions.go

```diff
@@ -96,6 +96,60 @@ type Positions interface {
 	Stop()
 }
 
+// LegacyFile is the copied struct for the static mode positions file.
+type LegacyFile struct {
+	Positions map[string]string `yaml:"positions"`
+}
+
+// ConvertLegacyPositionsFile will convert the legacy positions file to the new format if:
+// 1. There is no file at the newpath
+// 2. There is a file at the legacy path and that it is valid yaml
+// If all the above is true then the legacy file will be deleted.
+func ConvertLegacyPositionsFile(legacyPath, newPath string, l log.Logger) {
```

Comment:
nit: This one ended up a bit lengthy. Consider breaking it up a bit. For example, `readLegacyPositionsFile` could exist that is functionally similar to `readPositionsFile`